sonic-ycabled/ycable/ycable_utilities/y_cable_helper.py

```diff
@@ -4080,4 +4013,120 @@ def join(self):
  
             raise self.exc
 
+class GracefulRestartClient:
+    def __init__(self, port, channel: grpc.aio.secure_channel, read_side):
+        self.port = port
+        self.stub = linkmgr_grpc_driver_pb2_grpc.GracefulRestartStub(channel)
+        self.request_queue = asyncio.Queue()
+        self.response_queue = asyncio.Queue()
+        self.read_side = read_side
+
+    async def send_request_and_get_response(self):
+        while True:
+            tor = await self.request_queue.get()
+            request = linkmgr_grpc_driver_pb2.GracefulAdminRequest(tor=tor)
+            response = None 
+            try:
+                response_stream = self.stub.NotifyGracefulRestartStart(request)
+                index = 0
+                async for response in response_stream:
+                    helper_logger.log_notice("Async client received from direct read period port = {}: period = {} index = {} guid = {} notifytype {} msgtype = {}".format(self.port, response.period, index, response.guid, response.notifytype, response.msgtype))
+                    helper_logger.log_debug("Async Debug only :{} {}".format(dir(response_stream), dir(response)))
+                    index = index+1
+                    if response == grpc.aio.EOF:
+                        break
+                helper_logger.log_notice("Async client finished loop from direct read period port:{} ".format(self.port))
+                index = index+1
+            except grpc.RpcError as e:
+                helper_logger.log_notice("Async client port = {} exception occured because of {} ".format(self.port, e.code()))
+
+            await self.response_queue.put(response)
+
+    async def process_response(self):
```

Comment:
Will response ever be written DBs for other process to consume? Or the only action item is to sleep gRPC client for now?

Reply:
Currently the plan is to log it, if its going to be used later on for more purposes, we will put it to DB